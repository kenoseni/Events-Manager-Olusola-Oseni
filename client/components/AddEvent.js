import React, {Component} from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import SubmitButton from './SubmitButton';

class AddEvent extends Component {
  constructor(props) {
    super(props)
    this.state = {
      value: 'Choose a Center',
      event: {},
      centers: []
    }
    this.submit = this.submit.bind(this)
    this.getInput = this.getInput.bind(this)
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.eventCenters.centers !== this.props.eventCenters.centers) {
      this.setState({centers: nextProps.eventCenters.centers})
    }
  }
  
  getInput(e) {
    this.setState({ event: {
      ...this.state.event,
      [e.target.name]: e.target.value
    },
   })
  }
  
  submit(e) {
    e.preventDefault();
    const {name, centerId, date, time} = this.state.event
    $('#addEvent').modal('hide')
    this.props.addEvent({
      name,  
      centerId, 
      date, 
      time
    })
  }
  render() {
    const {centers} = this.state
    return (
      <div>
        <div className="container">            
          {/*<!-- Add Events Modal -->*/}
          <div className="modal fade" id="addEvent" role="dialog">
            <div className="modal-dialog modal-lg">      
              {/*<!-- Modal content-->*/}
              <div className="modal-content">
                <div className="modal-header">
                  <h4 className="modal-title text-center font-weight-bold">{this.props.title}</h4>
                  <button type="button" className="close" data-dismiss="modal">&times;</button>   
                </div>
                <div className="modal-body">
                  <form action="" method="" role="form" onSubmit={this.submit}>       
                    <div className="card">
                      <div className="card-body text-center">
                        <div className="input-group input-group-sm">
                          <span className="input-group-addon" id="sizing-addon1">Event Name</span>
                          <input type="text" name="name" onChange={this.getInput} className="form-control" aria-describedby="sizing-addon1" />
                        </div>
                      </div>
                      <div className="card-body">
                        <div className="container-fluid">
                          <div className="row">
                            <div className="col-xs-12">
                              <div className="input-group input-group-sm">
                                <span className="input-group-addon" id="sizing-addon1">Location</span>
                                <select className="form-control" name="centerId" onChange={this.getInput}>
                                  <option value={this.state.value}>Choose a Center</option>
                                  {
                                    centers.map(center => <option key={center.id} value={center.id}> {(center) ? center.name.toUpperCase(): ''} </option>)
                                  }
                                </select>
                              </div>
                            </div>
                            <div className="col-xs-12">
                              <div className="input-group input-group-sm">
                                <span className="input-group-addon" id="sizing-addon1">Date</span>
                                <input type="date" name="date" onChange={this.getInput} className="form-control" aria-describedby="sizing-addon1" />
                              </div>
                            </div>
                            <div className="col-xs-12">
                              <div className="input-group input-group-sm">
                                <span className="input-group-addon" id="sizing-addon1">Time</span>
                                <input type="time" name="time" onChange={this.getInput} className="form-control" aria-describedby="sizing-addon1" />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <br />
                    <SubmitButton name='ADD EVENT'/>
                  </form>            
                </div>
              </div> 
            </div>
          </div>
        </div>
      </div>
    )
  }
}
export default AddEvent;
