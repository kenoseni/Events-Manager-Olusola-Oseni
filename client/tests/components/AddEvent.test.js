import React from 'react';
import { configure, shallow } from 'enzyme';
import sinon from 'sinon';
import Adapter from 'enzyme-adapter-react-16';
import AddEvent from '../../components/AddEvent';

configure({ adapter: new Adapter() });

const mockaddEvent = jest.fn();
const mockisValid = jest.fn();
const mockaddUserEvent = jest.fn();
let props = {
  eventCenters: {
    allCenters: []
  }
};
const wrapper = shallow(<AddEvent addEvent={mockaddEvent} {...props} />);

describe('<AddEvent />', () => {
  it('should render a <AddEvent /> component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  // it('should call the isValid once', () => {
  //   const spy = sinon.spy(wrapper.instance(), 'isValid')
  //   wrapper.instance().isValid()
  //   expect(mockisValid).toHaveBeenCalledTimes(1)
  // });
  //   it('it should call getAllUsers function', () => {
  //     wrapper.setProps({
  //       eventCenters: {
  //         allCenters: [
  //           {
  //             id: 4,
  //             name : 'cincinati',
  //             description: 'you should knw',
  //             location: 'Lagos',
  //             price: '100000',
  //             capacity: 1000,
  //             image: ''
  //           }
  //         ]
  //       }
  //     })
  //     expect(wrapper.find('.alert').length).toBe(1);

  it('it should call addUserEvent function', () => {
    wrapper.state({
      value: 'Choose a Center',
      event: {},
      allCenters: [],
      error: {}
    });
    wrapper.setState({
      ...wrapper.state,
      event: {
        name: 'me',
        centerId: '1',
        startDate: '2018-07-15',
        endDate: '2018-07-16',
        time: '4 pm'
      }
    });
  });

  // const spy = sinon.spy(wrapper.instance(), 'addUserEvent');
  // wrapper.instance().addUserEvent();
  // expect(mockaddUserEvent).toHaveBeenCalledTimes(1);
});

// // describe('componenentWillReceiveProps', () => {
// //   it('should set a new state after receiving new props', () => {
// //     sinon.spy(wrapper, 'addUserEvent');
// //     wrapper.setProps({
// //       ...wrapper.props.eventCenters.allCenters,
// //       allCenters: ['big', 'small']
// //     });
// //     expect(addUserEvent).toBeCalled(1);
// //   });
// });
