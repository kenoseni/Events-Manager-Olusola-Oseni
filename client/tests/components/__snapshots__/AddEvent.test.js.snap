// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddEvent /> should render a <AddEvent /> component 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="modal fade right"
      id="addEvent"
      role="dialog"
    >
      <div
        className="modal-dialog modal-full-height modal-right"
      >
        <div
          className="modal-content"
        >
          <div
            className="modal-header"
            id="addEventModal"
          >
            <h4
              className="modal-title text-center font-weight-bold"
              id="addEventTitle"
            />
            <button
              className="close"
              data-dismiss="modal"
              type="button"
            >
              Ã—
            </button>
          </div>
          <div
            className="modal-body"
          >
            <form
              action=""
              method=""
              onSubmit={[Function]}
              role="form"
            >
              <div
                className=""
              >
                <div
                  className=""
                >
                  <div
                    className="input-group mb-3"
                  />
                </div>
                <div
                  className="container-fluid"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-12"
                      id="addEventForm"
                    >
                      <span
                        className=""
                        id="sizing-addon1"
                      >
                        Event Name:
                      </span>
                      <input
                        aria-describedby="sizing-addon1"
                        className="form-control"
                        id="eventName"
                        name="name"
                        onChange={[Function]}
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    className=""
                  >
                    <div
                      className="row"
                    >
                      <div
                        className="col-12"
                      >
                        <span
                          className=""
                          id="sizing-addon1"
                        >
                          Location:
                        </span>
                        <select
                          className="form-control"
                          id="eventLocation"
                          name="centerId"
                          onChange={[Function]}
                        >
                          <option
                            value="Choose a Center"
                          >
                            Choose a Center
                          </option>
                        </select>
                      </div>
                    </div>
                    <div
                      className="row"
                    >
                      <div
                        className="col-12"
                      >
                        <span
                          className=""
                          id="sizing-addon1"
                        >
                          Start Date:
                        </span>
                        <input
                          aria-describedby="sizing-addon1"
                          className="form-control"
                          id="startDate"
                          name="startDate"
                          onChange={[Function]}
                          type="date"
                        />
                      </div>
                    </div>
                    <div
                      className="row"
                    >
                      <div
                        className="col-12"
                      >
                        <span
                          className=""
                          id="sizing-addon1"
                        >
                          End Date:
                        </span>
                        <input
                          aria-describedby="sizing-addon1"
                          className="form-control"
                          id="endDate"
                          name="endDate"
                          onChange={[Function]}
                          type="date"
                        />
                      </div>
                    </div>
                    <div
                      className="row"
                    >
                      <div
                        className="col-12"
                      >
                        <span
                          className=""
                          id="sizing-addon1"
                        >
                          Time:
                        </span>
                        <input
                          aria-describedby="sizing-addon1"
                          className="form-control"
                          id="timeT"
                          name="time"
                          onChange={[Function]}
                          type="text"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <br />
              <SubmitButton
                id="submitButton"
                name="ADD EVENT"
              />
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
